typespec-extension/src/code-model-builder.ts

```diff
@@ -1440,7 +1443,24 @@ export class CodeModelBuilder {
 
         for (const [_, opParameter] of parameters.properties) {
           const serializedName = this.getSerializedName(opParameter);
-          const existParameter = op.parameters.find((it) => it.language.default.serializedName === serializedName);
+          const paramLocation = this.getParameterLocation(opParameter);
+          let existParameter: Parameter | undefined;
+          if (paramLocation === "BodyProperty") {
+            // property of body, it won't match existing parameter (whose paramLocation be body, path, query, header)
+            existParameter = undefined;
+          } else {
+            existParameter = op.parameters.find((it) => {
+              const sameParamLocation = paramLocation === it.protocol.http?.in;
+              const sameSerializedName = it.language.default.serializedName === serializedName;
+              if (paramLocation === ParameterLocation.Body) {
+                // body, same paramLocation, as there could only be 1 body in operation
+                return sameParamLocation;
+              } else {
+                // path, query, header, require same serializedName and same paramLocation
+                return sameParamLocation && sameSerializedName;
+              }
+            });
+          }
```

Comment:
We should be migrating to TCGC operation/method on this.

To be done after TCGC package migration.

Also, the "options bag" feature would also touch this part.